### ISSUE TITLE
HHVM Aborted! 

### ISSUE THREAD

**Issue by JasonXt** [facebook/hhvm#4718]
Ubuntu 14.04

Error Log:

[Tue Jan 27 15:25:31 2015] [hphp] [2318:7f5cdafff700:0:000001] [] FastCGI protocol: received an invalid record
[Tue Jan 27 15:25:31 2015] [hphp] [2318:7f5cdafff700:0:000002] [7f5ce54e8c30:7f5ce54e8bb9:7f5ce54ebfc8:7f5ce5df46b5:7f5ce5df2836:7f5ce5df2863:7f5ce5df333f:b2bf43:1fd434a:1fd3ede:1fd94fe:7f5ce9341f24:1fd6c57:1fd8415:1e5cdea:1e5e41a:7f5ce5e45bf0:7f5ce60a0182:7f5ce55acefd] Core dumped: Aborted
[Tue Jan 27 15:25:31 2015] [hphp] [2318:7f5cdafff700:0:000003] [7f5ce54e8c30:7f5ce54e8bb9:7f5ce54ebfc8:7f5ce5df46b5:7f5ce5df2836:7f5ce5df2863:7f5ce5df333f:b2bf43:1fd434a:1fd3ede:1fd94fe:7f5ce9341f24:1fd6c57:1fd8415:1e5cdea:1e5e41a:7f5ce5e45bf0:7f5ce60a0182:7f5ce55acefd] Stack trace in /tmp/stacktrace.2316.log

This is the stacktrace.2316.log:
```
Host: iZ28t8nlpldZ
ProcessID: 2318
ThreadID: 7f5cdafff700
ThreadPID: 2322
Name: unknown program
Type: Aborted
Runtime: hhvm
Version: tags/HHVM-3.4.2-0-g40d85d7386b3342ab848fc45c6892433c89ef5b2
DebuggerCount: 0


# 0  gsignal at /lib/x86_64-linux-gnu/libc.so.6:0
# 1  abort at /lib/x86_64-linux-gnu/libc.so.6:0
# 2  __gnu_cxx::__verbose_terminate_handler() at /usr/lib/x86_64-linux-gnu/libstdc++.so.6:0
# 3  std::rethrow_exception(std::__exception_ptr::exception_ptr) at /usr/lib/x86_64-linux-gnu/libstdc++.so.6:0
# 4  std::terminate() at /usr/lib/x86_64-linux-gnu/libstdc++.so.6:0
# 5  __cxa_pure_virtual at /usr/lib/x86_64-linux-gnu/libstdc++.so.6:0
# 6  std::_Function_handler<void (), HPHP::FastCGITransport::sendImpl(void const*, int, int, bool)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at fastcgi-transport.cpp:0
# 7  folly::EventBase::runFunctionPtr(std::function<void ()>*) at /usr/bin/hhvm:0
# 8  folly::EventBase::FunctionRunner::messageAvailable(std::pair<void (*)(void*), void*>&&) at /usr/bin/hhvm:0
# 9  folly::NotificationQueue<std::pair<void (*)(void*), void*> >::Consumer::handlerReady(unsigned short) at /usr/bin/hhvm:0
# 10 event_base_loop at /usr/lib/x86_64-linux-gnu/libevent-2.0.so.5:0
# 11 folly::EventBase::loopBody(int) at /usr/bin/hhvm:0
# 12 folly::EventBase::loopForever() at /usr/bin/hhvm:0
# 13 proxygen::WorkerThread::runLoop() at /usr/bin/hhvm:0
# 14 std::thread::_Impl<std::_Bind_simple<proxygen::WorkerThread::start()::{lambda()#1} ()> >::_M_run() at WorkerThread.cpp:0
# 15 std::this_thread::__sleep_for(std::chrono::duration<long, std::ratio<1l, 1l> >, std::chrono::duration<long, std::ratio<1l, 1000000000l> >) at /usr/lib/x86_64-linux-gnu/libstdc++.so.6:0
# 16 start_thread at /lib/x86_64-linux-gnu/libpthread.so.0:0
# 17 clone at /lib/x86_64-linux-gnu/libc.so.6:0
```

What's the problem?

**Comment by jwatzman** [comment]
Does this happen consistently? If so, what causes it to happen?

Is this the package for Ubuntu 14.04 from dl.hhvm.com, or did you compile HHVM yourself?

**Comment by paulbiss** [comment]
Our FastCGI implementation has quite a few pure-virtual functions, I suspect there is a destructor related race condition hiding somewhere. (See also #4068, #3759, and #3599)

**Comment by JasonXt** [comment]
@jwatzman  For recently ，it happens consistently  And I just update to v3.5 by apt get

**Comment by JasonXt** [comment]
@jwatzman what can I do? I am a new guy

**Comment by r3wt** [comment]
@JasonXt  you can find the previously working version from dl.hhvm. use wget to download the package, then run
`sudo apt-get autoremove hhvm -y`
then run
`sudo dpkg  -i path/to/downoadedpackage.deb`

**Comment by jwatzman** [comment]
@JasonXt If reverting to a previous version of HHVM would help, @r3wt posted good directions on how to install them. [In particular, old versions for Ubuntu are here.](http://dl.hhvm.com/ubuntu/pool/main/h/hhvm/)

To actually fix the problem, someone will need to reproduce the crash and dig into why it's happening. If you know how to get us a core dump with the `hhvm-dbg` package, that might be helpful. Information on what your app is in order to reproduce the issue is also helpful. Beyond that, it will just take someone who has the time to figure out what's going on.

**Comment by jwatzman** [comment]
Also @JasonXt what application are you running? Any information to help us reproduce the problem is helpful.

**Comment by jwatzman** [comment]
I am fairly confident that this is fixed by e13ca7d910554bd455980be1cac65d888d71a931 which will start showing up in nightly builds dated 2015.02.03, as well as in 3.6.0, 3.5.1, and 3.3.4 whenever those come out. I'm going to close this now; please reopen if you can repro in any of those versions.
